relay-server/src/actors/processor.rs

```diff
@@ -2689,7 +2689,16 @@ impl EnvelopeProcessorService {
             self.normalize_dsc(state);
             self.filter_event(state)?;
             self.run_dynamic_sampling(state);
-            self.extract_metrics(state)?;
+
+            let is_processing =
+                cfg!(feature = "processing") && self.inner.config.processing_enabled();
```

Comment:
You could simplify here. Enabling processing without the feature flag will throw errors during startup. Put another way, you can assume that `processing_enabled()` should never return `true` when not compiled with the relevant feature flags.

```suggestion
            let is_processing = self.inner.config.processing_enabled();
```

Reply:
makes sense, I just inlined it then